### PR TITLE
fix(add-app-overlay): disallow underscore in application name

### PR DESCRIPTION
**Pull request by vikram-raj**
Fix: https://github.com/openshiftio/openshift.io/issues/4414

### CODE REVIEW

File: src/app/space/add-app-overlay/add-app-overlay.component.spec.ts [hunk 1]
```diff
@@ -266,7 +266,7 @@ describe('AddAppOverlayComponent', () => {
 
     it('validate Project Name to be truthy', () => {
```
Comment (christianvogt) [inline src/app/space/add-app-overlay/add-app-overlay.component.spec.ts]: Test name no longer matches the test assertion.
You should keep a valid name test and move this assertion to the falsy test case.

Comment (joshuawilson) [reply src/app/space/add-app-overlay/add-app-overlay.component.spec.ts]: I fixed it. Please review.

**Comment by christianvogt** [comment]
@joshuawilson @vikram-raj I've updated the test cases accordingly due to the removal of the underscore character as a valid project name character

**Comment by joshuawilson** [comment]
I'm not going to close this yet, but I'm almost positive the UI is doing what it is supposed to.
We want to allow underscores in the middle.

I'll wait for the further discussion on the issue.

**Comment by joshuawilson** [comment]
I have confirmation on the main issue. Underscores are not allowed by kube/OpenShift.

File: src/app/space/add-app-overlay/add-app-overlay.component.ts [hunk 2]
```diff
@@ -141,9 +141,9 @@ export class AddAppOverlayComponent implements OnInit, OnDestroy {
    * @returns boolean
    */
   isValidProjectName(projectName: string): boolean {
-    // allows only '-', '_' and 4-40 characters (must start with alphabetic and end with alphanumeric)
-    // no continuous '-' or '_' is allowed
-    const pattern = /^[a-zA-Z](?!.*--)(?!.*__)[a-zA-Z0-9-_]{2,38}[a-zA-Z0-9]$/;
+    // allows only '-' and 4-40 characters (must start with alphabetic and end with alphanumeric)
+    // no continuous '-' is allowed
+    const pattern = /^[a-zA-Z](?!.*--)[a-zA-Z0-9-]{2,38}[a-zA-Z0-9]$/;
```
Comment (joshuawilson) [inline src/app/space/add-app-overlay/add-app-overlay.component.ts]: The spec does not allow for upper case letters either. Please update accordingly.

File: src/app/space/add-app-overlay/add-app-overlay.component.spec.ts [hunk 3]
```diff
@@ -60,8 +60,8 @@ describe('AddAppOverlayComponent', () => {
       });
     },
     validateProjectName(projectName: string): boolean {
-      // allows only '-', '_', ' ' and 4-40 characters (must start and end with alphanumeric)
-      const pattern = /^[a-zA-Z0-9][a-zA-Z0-9-_\s]{2,38}[a-zA-Z0-9]$/;
+      // allows only '-', ' ' and 4-40 characters (must start and end with alphanumeric)
+      const pattern = /^[a-zA-Z0-9][a-zA-Z0-9-\s]{2,38}[a-zA-Z0-9]$/;
```
Comment (joshuawilson) [inline src/app/space/add-app-overlay/add-app-overlay.component.spec.ts]: The spec does not allow for upper case letters either. Please update accordingly.

File: src/app/space/add-app-overlay/add-app-overlay.component.ts (hunk 2 above)
Comment (vikram-raj) [reply src/app/space/add-app-overlay/add-app-overlay.component.ts]: Updated the regex here https://github.com/fabric8-ui/fabric8-ui/pull/3407/commits/0cf747190aff413b812511016bb306e0bc156d7b.

File: src/app/space/add-app-overlay/add-app-overlay.component.spec.ts (hunk 3 above)
Comment (vikram-raj) [reply src/app/space/add-app-overlay/add-app-overlay.component.spec.ts]: Updated the regex here https://github.com/fabric8-ui/fabric8-ui/pull/3407/commits/0cf747190aff413b812511016bb306e0bc156d7b.

**Pull request merged**